### ISSUE TITLE
Editing goes wacky after Updating to Version 1.1.0

### ISSUE THREAD

**Issue by darkmunn** [Automattic/simplenote-electron#671]
#### Steps to reproduce
Tried to reproduce after closing and not able to, yet!

#### What I expected
Characters to appear where the cursor is shown while typing

#### What happened instead
While typing the cursor jumped to the beginning of the line, but yet while continue to type the characters were appearing where the cursor had been prior to jumping sporadically to the first position on the same line.

Used the mouse to put the cursor on a different line, but as I typed the characters continued to appear where the problem first appeared, several lines down.

#### OS version
Windows 10

#### Notes
Because I use your app daily I will keep you updated when this happens again and possibly how to reproduce the problem.

Really enjoy the app and have for a few years now!

**Comment by dmsnell** [comment]
Thanks @darkmunn for reporting - that sounds frustrating! Can you help us out a little more? Does it do this reliably every time you use it? Are there only certain times it does it? If you start typing and continue is that different than starting, pausing for a certain time, then typing again?

**Comment by darkmunn** [comment]
So far I can’t get it to repeat the issue after I closed and reopened the program.  I’m hoping it was only after the update and possibly closing it somehow eliminated the issue.  Because I’m in the app a lot I will report it as I find out more.

**Comment by PhiLhoSoft** [comment]
Curiously, I actually saw this behavior (for the first time) a few days ago, before I upgrade. So it was in 1.0.8 on Win 10 too. I typed some text in the middle of a rather long node, and the caret went to the end of the note in the middle of my typing (on sync ?).
It happened a few times, rather randomly.
Sorry, that's a pretty vague bug report, I fear...
I can't reproduce the problem with 1.1.0...

**Comment by rachelsquirrel** [comment]
I've received a similar report of random characters appearing in a user's note that the user did not type themselves. This happened after the update to 1.1.0.  

#804100-zen

**Comment by jamiepalatnik** [comment]
Another report here: 821993-zen

**Comment by dmsnell** [comment]
Hopefully fixed when #692 is ready.

**Comment by zandyring** [comment]
It sounds like this is what's happening in 1009130-zen as well.

**Comment by natanielcz** [comment]
@dmsnell can You tell me where are reported those `*-zen` bugs?

**Comment by dmsnell** [comment]
@natanielcz sadly you won't be able to get access to those logs.

I believe I have pinned-down pretty much the cause of this though and those reports and this one are all facets of the bug. The data flow is isn't quite proper and that's introducing races between the value of the note on the Simplenote servers, the value of the note in IndexedDB, the value of the note in Red memory, and the value of the note in the local state for the editor component.

We need to do one of several things:
 - Move the debouncing of sending updates deeper into the Simperium library and let the app "always have the state of truth"
 - Treat the editor fully as a viewer of updatable data and merge updates from the Simperium library into the current session
 - Something else

Either way we have to merge the updates into the cursor position. When we currently get updates we end up replacing the entire contents of the editor and that resets the cursor position - not good.

**Comment by ChrissiePollock** [comment]
Similar issue in #1220599-Zen

> Sometimes when typing, the cursor would jump to the bottom of a note. Like when I add or change some text in the middle of a note, it would suddenly lose its place and appear at the bottom after the very last character. I'd have to put it back to continue typing where I need. Happens with longer notes, like 3000 characters (though, I'd think it's not too huge). I'm pretty sure it's not my fingers or anything like that, otherwise it'd happen in other notepads and text editors. Also, it have happened enough times to be sure it's Simplenote's glitch.

**Comment by mirka** [comment]
While there are still underlying constraints that could cause this to happen in edge cases, users should be experiencing much fewer of these glitches, due to improvements made in #1193.

I'm closing this for being outdated, but feel free to post a new issue if you encounter this again.

**Comment by iamgabrielma** [comment]
@mirka I seem to have another case here: 1812291-zen 

> when I move my mouse cursor to a specific place in a text line, the cursor suddenly jumps to the beginning of the line or even to another line as I start typing my text.
> In addition, the undo function does not seem to work in this case.

I understand this should be fixed with v1.4.1 so it may be one of these edge cases you mention. Anything specific we could ask them for in order to check where this may be coming from?

**Comment by mirka** [comment]
@iamgabrielma The currently known edge case is #1134. Basically,

1. Place cursor somewhere in the note.
2. Invoke the Preferences dialog from the menu or <kbd>Ctrl+,</kbd>
3. Close the Preferences dialog.
4. Cursor jumps.

If it is not this case, you could ask for example: Is there a certain line that always triggers it? If not a specific line, do the triggering lines seem to share a similarity? (Indented, Markdown list, particular language, etc.)
